bitmapcontainer_test.go

```diff
@@ -485,3 +485,50 @@ func TestBitmapcontainerNextHasMany(t *testing.T) {
 		assert.Equal(t, 256, result)
 	})
 }
+
+func TestBitmapcontainerOrArrayCardinality(t *testing.T) {
+	t.Run("Empty Bitmap and Empty Array", func(t *testing.T) {
+		array := newArrayContainer()
+		bc := newBitmapContainer()
+		result := bc.orArrayCardinality(array)
+
+		assert.Equal(t, 0, result)
+	})
+
+	t.Run("Populated Bitmap with Empty Array", func(t *testing.T) {
```

Comment:
@lemire  This test is failing. Seems like a bug. The following two test compare that behavior with other containers.

Reply:
Indeed. It looks like you caught a bug.

Would you try the following fix at https://github.com/RoaringBitmap/roaring/blob/5aca9677d8270ffec04f4535b092216796de2135/bitmapcontainer.go#L544

```Go
func (bc *bitmapContainer) orArrayCardinality(value2 *arrayContainer) int {
	answer := bc.getCardinality()
	c := value2.getCardinality()
	for k := 0; k < c; k++ {
		// branchless:
		v := value2.content[k]
		i := uint(v) >> 6
		bef := bc.bitmap[i]
		aft := bef | (uint64(1) << (v % 64))
		answer += int((bef - aft) >> 63)
	}
	return answer
```